pkg/kubelet/kubelet.go

```diff
@@ -1559,18 +1598,26 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 		return syncErr
 	}
 
+	podAdmitSpan.End()
+
+	ctx, networkConditionSpan := trace.StartSpan(ctx, "Network condition check")
 	// If the network plugin is not ready, only start the pod if it uses the host network
 	if rs := kl.runtimeState.networkErrors(); len(rs) != 0 && !kubecontainer.IsHostNetworkPod(pod) {
+		networkConditionSpan.Annotate([]trace.Attribute{}, "Network errors detected, cannot proceed")
 		kl.recorder.Eventf(pod, v1.EventTypeWarning, events.NetworkNotReady, "%s: %v", NetworkNotReadyErrorMsg, rs)
 		return fmt.Errorf("%s: %v", NetworkNotReadyErrorMsg, rs)
 	}
+	networkConditionSpan.End()
```

Comment:
Probably here you need the ctx to also revert the context to the previous value, otherwise the next span cgroupSpan will be a child of the networkConditionSpan. Same everywhere you have this problem.